File: backend/src/main/java/pt/isel/rugby/model/TrainingSchedule.java
package pt.isel.rugby.model;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "training_schedule")
@Data
public class TrainingSchedule implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private String description;

    @Column
    private String link;

    @Column
    private Date date;

    @ManyToMany(mappedBy = "trainingSchedules")
    private List<Athlete> athletes;
}

File: frontend2/Rugby/src/app/classes/trainingschedule.ts
import {Athlete} from './athlete';

export class TrainingSchedule {
  constructor(
    public id?: number,
    public description?: string,
    public link?: string,
    public date?: Date,
    public athletes?: Athlete[]
  ) {
    this.id = id ? id : 0;
    this.description = description ? description : '';
    this.link = link ? link : '';
    this.date = date ? date : new Date(0);
    this.athletes = athletes ? athletes : [];
  }
}
